incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -517,6 +554,42 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		if err := r.client.Update(r.ctx, vwh); err != nil {
 			return reconcile.Result{Requeue: true}, err
 		}
+
+		// Ensure certs on CRDs.
+		if result, err := r.ensureCRDCerts(artifacts.CertPEM); err != nil {
+			return result, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileWH) ensureCRDCerts(certPem []byte) (reconcile.Result, error) {
+	for _, crdName := range r.crdNames {
+		crd := &unstructured.Unstructured{}
+		crd.SetGroupVersionKind(crdGVK)
+		crdKey := types.NamespacedName{Name: crdName}
+		if err := r.client.Get(r.ctx, crdKey, crd); err != nil {
+			if k8sErrors.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{Requeue: true}, err
+		}
+
+		if !crd.GetDeletionTimestamp().IsZero() {
+			return reconcile.Result{}, nil
+		}
+
+		crLog.Info("ensuring CA cert on CRD conversion webhook")
+		if err := injectCertToConversionWebhook(crd, certPem); err != nil {
+			crLog.Error(err, "unable to inject cert to CRD conversion webhook")
+			return reconcile.Result{}, err
```

Comment:
@adrianludwin since here we don't want to enqueue the request again if there is an error in the injection (e.g. cannot find the `caBundle` field), I still keep this `ensureCRDCerts` func to return both result and error.

Reply:
I'm not sure I understand? Returning a non-nil error guarantees that we'll re-enqueue the request.

Reply:
And I think that's fine. Maybe the `caBundle` field was missing due to an operator error but it gets added later, in which case this is self-repaired (good). Otherwise, we continue trying this every few seconds and putting the error in the log each time, which means that when someone looks at the log, the error message will always be fresh (also good). So re-enqueuing is usually a very good idea :)